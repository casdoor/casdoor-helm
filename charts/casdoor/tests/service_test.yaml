suite: test service
templates:
- service.yaml

tests:
- it: should render with default values
  asserts:
  - matchSnapshot: {}

- it: should set service type to NodePort
  set:
    service.type: NodePort
  asserts:
  - matchSnapshot: {}

- it: should enable ldap port
  set:
    ldap.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should set custom service port
  set:
    service.port: 7979
  asserts:
  - matchSnapshot: {}

- it: should set nodePort when service type is NodePort
  set:
    service.type: NodePort
    service.nodePort: 30000
  asserts:
  - matchSnapshot: {}
