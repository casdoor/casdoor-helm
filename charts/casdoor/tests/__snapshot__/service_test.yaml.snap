should enable ldap port:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ports:
        - name: http
          port: 8000
          protocol: TCP
          targetPort: http
        - name: ldap
          port: 389
          protocol: TCP
          targetPort: 10389
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: casdoor
      type: ClusterIP
should render with default values:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ports:
        - name: http
          port: 8000
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: casdoor
      type: ClusterIP
should set custom service port:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ports:
        - name: http
          port: 7979
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: casdoor
      type: ClusterIP
should set nodePort when service type is NodePort:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ports:
        - name: http
          nodePort: 30000
          port: 8000
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: casdoor
      type: NodePort
should set service type to NodePort:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ports:
        - name: http
          port: 8000
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: casdoor
      type: NodePort
